src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -1163,7 +1163,7 @@ protected void UpdateRuleSetError(IRuleSetFile ruleSetFile)
             }
         }
 
-        protected void SetOutputPathAndRelatedData(string objOutputPath, bool hasSameBinAndObjOutputPaths = false)
+        protected void SetOutputPathAndRelatedData(string objOutputPath, string binOutputPath)
```

Comment:
This method is really confusing it requires you to pass in the bin and obj (which you pass in the fields), but then goes and reads some of the fields? 

If I was to design this, I would set the required fields when they change and then call a UpdateOutputPathAndRelatedData() that took zero arguments that read the fields. It feels a lot less confusing as to where the data is coming from.